### ISSUE TITLE
NullReferenceException when iterating DocumentQuery in IIS on V2.9.0

### ISSUE THREAD

**Issue by bcrobinson** [Azure/azure-cosmos-dotnet-v2#750]
**Describe the bug**
After updating the `Microsoft.Azure.DocumentDB.Core` package to 2.9.0 we started getting null reference exceptions when querying document collections.
After downgrading to V2.8.1 and the exception went away.

**To Reproduce**
For example the following code will throw an error when running in an Asp.Net website in IIS or IIS Express

```csharp 
 var items = documentClient
    .CreateDocumentQuery<Thing>(collectionUri)
    .Where(e => e.Somthing == "abc")
    .ToArray();
```
**Actual behavior**
The method throws the exception:

```
Type:System.AggregateException
Message:One or more errors occurred.
Aggregate Exception with 1 inner exceptions

Stack Trace:
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Microsoft.Azure.Documents.Linq.DocumentQuery`1.<GetEnumerator>d__31.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at DataGateway.Server.Document.AccountDocumentService.<GetAccounts>d__5.MoveNext()
```

With the following inner exception:

```
Message:Object reference not set to an instance of an object.
Stack Trace:
   at Microsoft.Azure.Documents.ServiceInteropWrapper.<>c.<.cctor>b__9_0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Azure.Documents.CustomTypeExtensions.ByPassQueryParsing()
   at Microsoft.Azure.Documents.Query.DocumentQueryExecutionContextFactory.<CreateDocumentQueryExecutionContextAsync>d__3.MoveNext()
```

**Environment summary**
`Microsoft.Azure.DocumentDB.Core` 2.9.0
Running in an Asp.Net Web Api (_not core_) website, both IIS Express localy & IIS on an Azure Web App

**Additional context**

Decompiling the 2.9.0 libs the `Microsoft.Azure.Documents.ServiceInteropWrapper` has the following code in it (which is the `.<>c.<.cctor>b__9_0` method):

```csharp
internal static Lazy<bool> AssembliesExist = new Lazy<bool>(delegate
{
    if (!IsGatewayAllowedToParseQueries())
    {
        return true;
    }
    DefaultTrace.TraceInformation("Assembly location: " + typeof(ServiceInteropWrapper).GetTypeInfo().get_Assembly().Location);
    if (typeof(ServiceInteropWrapper).GetTypeInfo().get_Assembly().IsDynamic)
    {
        return true;
    }
    string directoryName = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
    string[] array = new string[1]
    {
        "Microsoft.Azure.Documents.ServiceInterop.dll"
    };
    foreach (string path in array)
    {
        string text = Path.Combine(directoryName, path);
        if (!File.Exists(text))
        {
            DefaultTrace.TraceVerbose("ServiceInteropWrapper assembly not found at " + text);
            return false;
        }
    }
    return true;
});
```

If I run `Assembly.GetEntryAssembly()` inside an Asp.Net website it returns null, it looks like the line `string directoryName = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);` is causing the exception.

If I look at V2.8.1 (which doesn't cause the exception) that line is instead:
`string directoryName = Path.GetDirectoryName(typeof(ServiceInteropWrapper).GetTypeInfo().get_Assembly().Location);`

**Comment by ausfeldt** [comment]
hotfix on the way

**Comment by ausfeldt** [comment]
Just released 2.9.1 with the hotfix.
